plugins/modules/network/routeros/routeros_api.py

```diff
@@ -0,0 +1,491 @@
+         after the command we can add C(.id).
+         Example path C(system script) and cmd C(run .id=*03)
+         is equivalent in RouterOS CLI C(/system script run number=0),
+          example path C(ip address) and cmd C(print)
+         equivalent in RouterOS CLI C(/ip address print).
+    type: str
+'''
+
+EXAMPLES = '''
+---
+- name: Test routeros_api
+  hosts: localhost
+  gather_facts: no
+  vars:
+    hostname: "ros_api_hostname/ip"
+    username: "admin"
+    password: "secret_password"
+
+    path: "ip address"
+
+    nic: "ether2"
+    ip1: "1.1.1.1/32"
+    ip2: "2.2.2.2/32"
+    ip3: "3.3.3.3/32"
+
+  tasks:
+    - name: Get "{{ path }} print"
+      community.network.routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+      register: print_path
+
+    - name: Result "{{ path }} print"
+      ansible.builtin.debug:
+        msg: '{{ print_path }}'
+
+    - name: Add ip address "{{ ip1 }}" and "{{ ip2 }}"
+      community.network.routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+        add: "{{ item }}"
+      loop:
+        - "address={{ ip1 }} interface={{ nic }}"
+        - "address={{ ip2 }} interface={{ nic }}"
+      register: addout
+
+    - name: Result routeros '.id' for new added items
+      ansible.builtin.debug:
+        msg: '{{ addout }}'
+
+    - name: Query for ".id" in "{{ path }} WHERE address == {{ ip2 }}"
+      community.network.routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+        query: ".id address WHERE address == {{ ip2 }}"
+      register: queryout
+
+    - name: Result query result and set fact with '.id' for {{ ip2 }}
+      ansible.builtin.debug:
+        msg: '{{ queryout }}'
+
+    - ansible.builtin.set_fact:
+        query_id : "{{ queryout['msg'][0]['.id'] }}"
+
+    - name: Update ".id = {{ query_id }}" taken with custom fact "fquery_id"
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+        update: ".id={{ query_id }} address={{ ip3 }}"
+      register: updateout
+
+    - name: Result prunt update status
+      ansible.builtin.debug:
+        msg: '{{ updateout }}'
+
+    - name: Remove ips -  stage 1 - query for '.id' "{{ ip2 }}" and "{{ ip3 }}"
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+        query: ".id address WHERE address == {{ item }}"
+      register: id_to_remove
+      loop:
+        - "{{ ip2 }}"
+        - "{{ ip3 }}"
+
+    # set fact for '.id' from 'query for {{ path }}'
+    - ansible.builtin.set_fact:
```

Comment:
```suggestion
    - name: "set fact for '.id' from 'query for {{ path }}'"
      ansible.builtin.set_fact:
```
i'm not sure the double quotes are needed here

Reply:
fixed